### PR TITLE
Missing idd defaults for CoolingTower:*Speed

### LINKED ISSUE #9262: Missing idd defaults for CoolingTower:*Speed
Issue overview
--------------
```
  A6 , \field Evaporation Loss Mode
       \type choice
       \key LossFactor
       \key SaturatedExit
```

```
  A7 , \field Blowdown Calculation Mode
       \type choice
       \key ConcentrationRatio
       \key ScheduledRate
```

Missing `\default SaturatedExit` and `\default ConcentrationRatio`, respectively, for:
* `CoolingTowerSingleSpeed`
* `CoolingTowerTwoSpeed`
* `CoolingTowerVariableSpeed`
* `CoolingTower:VariableSpeed:Merkel`

Also:
* `Performance Input Method` for `EvaporativeFluidCooler:SingleSpeed` (required-field)

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by Nigusse**
Pull request overview
---------------------
 - Fixes #9262
 - Adds default values to `Evaporation Loss Mode` and `Blowdown Calculation Mode` input fields for single speed, two speed, variable speed and variable speed Merkel cooling tower models. Extended the `Min-Fields` to all four cooling tower objects,  added default value of `0.008%` to the variable speed cooling tower `Drift Loss Percent` input field and changed the default value of `Cell Control` input field from `MinimalCell` to `MaximamCell` in the IDD for all four cooling tower models.

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [x] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [x] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [x] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [x] If feature, test running new feature, try creative ways to break it
 - [x] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: idd/Energy+.idd.in [hunk 1]
```diff
@@ -74491,6 +74491,7 @@ CoolingTower:SingleSpeed,
        \type choice
        \key LossFactor
        \key SaturatedExit
+       \default SaturatedExit
```
Comment (Nigusse) [inline idd/Energy+.idd.in:74494]: Added the default method for Evaporation Loss Mode

File: idd/Energy+.idd.in [hunk 2]
```diff
@@ -74509,6 +74510,7 @@ CoolingTower:SingleSpeed,
        \type choice
        \key ConcentrationRatio
        \key ScheduledRate
+       \default ConcentrationRatio
```
Comment (Nigusse) [inline idd/Energy+.idd.in:74513]: Added the default method for Blowdown Calculation Mode

File: idd/Energy+.idd.in [hunk 3]
```diff
@@ -74806,6 +74808,7 @@ CoolingTower:TwoSpeed,
        \type choice
        \key LossFactor
        \key SaturatedExit
+       \default SaturatedExit
```
Comment (Nigusse) [inline idd/Energy+.idd.in:74811]: Added the default method for Evaporation Loss Mode

File: src/EnergyPlus/CondenserLoopTowers.cc [hunk 4]
```diff
@@ -237,6 +237,7 @@ namespace CondenserLoopTowers {
 
         constexpr std::array<std::string_view, static_cast<int>(EvapLoss::Num)> EvapLossNamesUC{"LOSSFACTOR", "SATURATEDEXIT"};
         constexpr std::array<std::string_view, static_cast<int>(PIM::Num)> PIMNamesUC{"NOMINALCAPACITY", "UFACTORTIMESAREAANDDESIGNWATERFLOWRATE"};
+        constexpr std::array<std::string_view, static_cast<int>(Blowdown::Num)> BlowDownNamesUC = {"CONCENTRATIONRATIO", "SCHEDULEDRATE"};
```
Comment (Nigusse) [inline src/EnergyPlus/CondenserLoopTowers.cc:240]: Added BlowDown method lookup

File: src/EnergyPlus/CondenserLoopTowers.cc [hunk 5]
```diff
@@ -416,9 +417,6 @@ namespace CondenserLoopTowers {
 
             // begin water use and systems get input
             tower.EvapLossMode = static_cast<EvapLoss>(getEnumerationValue(EvapLossNamesUC, UtilityRoutines::MakeUPPERCase(AlphArray(6))));
-            if (AlphArray(6).empty()) {
-                tower.EvapLossMode = EvapLoss::MoistTheory;
-            }
```
Comment (Nigusse) [inline src/EnergyPlus/CondenserLoopTowers.cc:421]: The idd has default, so this code is not needed.

File: src/EnergyPlus/CondenserLoopTowers.cc [hunk 6]
```diff
@@ -432,25 +430,11 @@ namespace CondenserLoopTowers {
...
-                if ((NumNums < 21) && (tower.ConcentrationRatio == 0.0)) {
-                    // assume Concetratino ratio was omitted and should be defaulted
-                    tower.ConcentrationRatio = 3.0;
-                }
-            } else {
-                ShowSevereError(state, cCurrentModuleObject + '=' + AlphArray(1));
-                ShowContinueError(state, "Invalid, " + state.dataIPShortCut->cAlphaFieldNames(7) + " = " + AlphArray(7));
-                ErrorsFound = true;
-            }
+            tower.BlowdownMode = static_cast<Blowdown>(getEnumerationValue(BlowDownNamesUC, UtilityRoutines::MakeUPPERCase(AlphArray(7))));
```
Comment (Nigusse) [inline src/EnergyPlus/CondenserLoopTowers.cc:428]: Now uses getEnumerationValue, so the above code section is not needed.

File: src/EnergyPlus/CondenserLoopTowers.cc [hunk 7]
```diff
@@ -432,25 +430,11 @@ namespace CondenserLoopTowers {
...
             tower.SchedIDBlowdown = ScheduleManager::GetScheduleIndex(state, AlphArray(8));
             if ((tower.SchedIDBlowdown == 0) && (tower.BlowdownMode == Blowdown::Schedule)) {
-                ShowSevereError(state, cCurrentModuleObject + '=' + AlphArray(1));
-                ShowContinueError(state, "Invalid, " + state.dataIPShortCut->cAlphaFieldNames(8) + " = " + AlphArray(8));
+                ShowSevereError(state, format("Invalid, {} = \"{}\"", state.dataIPShortCut->cAlphaFieldNames(8), AlphArray(8)));
+                ShowContinueError(state, format("Entered in {} = \"{}\"", cCoolingTower_SingleSpeed, tower.Name));
                 ErrorsFound = true;
```
Comment (Nigusse) [inline src/EnergyPlus/CondenserLoopTowers.cc:433]: Updated the error message format.

File: src/EnergyPlus/CondenserLoopTowers.cc [hunk 8]
```diff
@@ -808,25 +789,11 @@ namespace CondenserLoopTowers {
...
-                tower.BlowdownMode = Blowdown::Concentration;
-                if ((NumNums < 29) && (tower.ConcentrationRatio == 0.0)) {
-                    // assume concentration ratio was omitted and should be defaulted
-                    tower.ConcentrationRatio = 3.0;
-                }
-            } else {
-                ShowSevereError(state, cCurrentModuleObject + '=' + AlphArray(1));
-                ShowContinueError(state, "Invalid " + state.dataIPShortCut->cAlphaFieldNames(7) + '=' + AlphArray(7));
-                ErrorsFound = true;
-            }
```
Comment (Nigusse) [inline src/EnergyPlus/CondenserLoopTowers.cc:825]: This code section is not needed. GetEnumerationValue() is used instead. The input field also has default value.

File: src/EnergyPlus/CondenserLoopTowers.cc [hunk 9]
```diff
@@ -1456,39 +1423,19 @@ namespace CondenserLoopTowers {
...
-                tower.EvapLossMode = EvapLoss::UserFactor;
-            } else if (UtilityRoutines::SameString(AlphArray(8), "SaturatedExit")) {
-                tower.EvapLossMode = EvapLoss::MoistTheory;
-            } else if (state.dataIPShortCut->lAlphaFieldBlanks(8)) {
-                tower.EvapLossMode = EvapLoss::MoistTheory;
-            } else {
-                ShowSevereError(state, "Invalid " + state.dataIPShortCut->cAlphaFieldNames(8) + '=' + AlphArray(8));
-                ShowContinueError(state, "Entered in " + cCurrentModuleObject + '=' + AlphArray(1));
-                ErrorsFound = true;
-            }
```
Comment (Nigusse) [inline src/EnergyPlus/CondenserLoopTowers.cc:1469]: Same as above.

File: src/EnergyPlus/CondenserLoopTowers.cc [hunk 10]
```diff
@@ -1456,39 +1423,19 @@ namespace CondenserLoopTowers {
...
-                tower.BlowdownMode = Blowdown::Schedule;
-            } else if (UtilityRoutines::SameString(AlphArray(9), "ConcentrationRatio")) {
-                tower.BlowdownMode = Blowdown::Concentration;
-            } else if (state.dataIPShortCut->lAlphaFieldBlanks(9)) {
-                tower.BlowdownMode = Blowdown::Concentration;
-            } else {
-                ShowSevereError(state, "Invalid " + state.dataIPShortCut->cAlphaFieldNames(9) + '=' + AlphArray(9));
-                ShowContinueError(state, "Entered in " + cCurrentModuleObject + '=' + AlphArray(1));
-                ErrorsFound = true;
-            }
```
Comment (Nigusse) [inline src/EnergyPlus/CondenserLoopTowers.cc:1487]: Same as above

File: src/EnergyPlus/CondenserLoopTowers.cc [hunk 11]
```diff
@@ -1770,25 +1707,11 @@ namespace CondenserLoopTowers {
...
-                tower.BlowdownMode = Blowdown::Concentration;
-                if ((NumNums < 25) && (tower.ConcentrationRatio == 0.0)) {
-                    // assume concentration ratio was omitted and should be defaulted
-                    tower.ConcentrationRatio = 3.0;
-                }
-            } else {
-                ShowSevereError(state, cCurrentModuleObject + '=' + AlphArray(1));
-                ShowContinueError(state, "Invalid " + state.dataIPShortCut->cAlphaFieldNames(11) + '=' + AlphArray(11));
-                ErrorsFound = true;
-            }
```
Comment (Nigusse) [inline src/EnergyPlus/CondenserLoopTowers.cc:1787]: Same as above.

File: src/EnergyPlus/CondenserLoopTowers.cc [hunk 12]
```diff
@@ -1747,17 +1694,7 @@ namespace CondenserLoopTowers {
...
-                tower.EvapLossMode = EvapLoss::UserFactor;
-            } else if (UtilityRoutines::SameString(AlphArray(10), "SaturatedExit")) {
-                tower.EvapLossMode = EvapLoss::MoistTheory;
-            } else if (state.dataIPShortCut->lAlphaFieldBlanks(10)) {
-                tower.EvapLossMode = EvapLoss::MoistTheory;
-            } else {
-                ShowSevereError(state, cCurrentModuleObject + '=' + AlphArray(1));
-                ShowContinueError(state, "Invalid " + state.dataIPShortCut->cAlphaFieldNames(10) + '=' + AlphArray(10));
-                ErrorsFound = true;
-            }
```
Comment (Nigusse) [inline src/EnergyPlus/CondenserLoopTowers.cc:1760]: Same as above.

File: tst/EnergyPlus/unit/CondenserLoopTowers.unit.cc [hunk 13]
```diff
@@ -3994,4 +3994,263 @@ TEST_F(EnergyPlusFixture, VSCoolingTowers_WaterOutletTempTest)
...
+    EXPECT_EQ(vSpdMerkel_tower.ConcentrationRatio, 3.0);
+    EXPECT_EQ(vSpdMerkel_tower.NumCell, 1);
+    EXPECT_EQ(vSpdMerkel_tower.MinFracFlowRate, 0.33);
+    EXPECT_EQ(vSpdMerkel_tower.MaxFracFlowRate, 2.50);
+    EXPECT_EQ(vSpdMerkel_tower.CellCtrl_Num, CondenserLoopTowers::CellCtrl::MaxCell);
+    EXPECT_EQ(vSpdMerkel_tower.EndUseSubcategory, "General");
+    EXPECT_EQ(vSpdMerkel_tower.PerformanceInputMethod_Num, CondenserLoopTowers::PIM::NominalCapacity);
+    EXPECT_EQ(vSpdMerkel_tower.HeatRejectCapNomCapSizingRatio, 1.25);
+
+}
```
Comment (Nigusse) [inline tst/EnergyPlus/unit/CondenserLoopTowers.unit.cc:4253]: The unit test checks for default values for each the four cooling tower models.

**Review by Nigusse — COMMENTED** [review]
Added default values to `Evaporation Loss Mode` and `Blowdown Calculation Mode` input fields of the four cooling tower models. Also cleaned up the `GetTowerInput` function related to these two input fields.

**Comment by Nigusse** [comment]
I see a couple of issues that I would like to document here and would like to get inputs from others if you want them to be addressed here: 
- The `Cell Control` input field default value all the four cooling tower models in the `IDD` is `MinimalCell` but in the code it is set to `MaximalCell` [here](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/CondenserLoopTowers.cc#L519-L540) for single speed cooling tower and [here](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/CondenserLoopTowers.cc#L1512-L1533) for variable speed cooling tower. **The default value in the IDD needs to change to  MaximalCell to match what is in the code.**
- All cooling tower except the variable speed has a default value of `0.008%` for `Drift Loss Percent` input field. The variable speed cooling tower has no default value in the idd and assumes 0.0 if this input field is left blank.

**Comment by Nigusse** [comment]
The warnings in 30 example files are due to the two additional fields with default. This is expected diffs in tabular output files as shown below:
image

**Comment by Nigusse** [comment]
All the 31 example files with warning messages and 171 example files failed are due to addition of two default fields and extension of the `Min-Fields` to the four cooling tower objects. All the diffs are in `eplustbl.htm` files only. Diffs summary:
- 31 example files have `Table small diffs` only. Now there are two input `Fields with Defaults` added as shown below. In these example files these two fields are specified in cooling tower object:

image

- 171 example files have `Table big diffs` only. The cooling tower object in these example files have two input `Fields with Defaults` and these fields are not specified in the idf and the input processor filled in the default values.

image

**Comment by Nigusse** [comment]
This PR is ready for review.

File: idd/Energy+.idd.in [hunk 14]
```diff
@@ -74316,7 +74316,7 @@ ChillerHeaterPerformance:Electric:EIR,
 
 !*****************CONDENSING EQUIPMENT*********************
 CoolingTower:SingleSpeed,
-       \min-fields 16
+       \min-fields 28
```
Comment (Nigusse) [inline idd/Energy+.idd.in:74319]: Increased the `Min-Fields` such the input processor fills in additional default values.

File: src/EnergyPlus/CondenserLoopTowers.cc [hunk 15]
```diff
@@ -1747,48 +1683,19 @@ namespace CondenserLoopTowers {
...
+            tower.EvapLossMode = static_cast<EvapLoss>(getEnumerationValue(EvapLossNamesUC, UtilityRoutines::MakeUPPERCase(AlphArray(10))));
 
             tower.UserEvapLossFactor = NumArray(23);        //  N23 , \field Evaporation Loss Factor
             tower.DriftLossFraction = NumArray(24) / 100.0; //  N24, \field Drift Loss Percent
-            if ((NumNums < 24) && (tower.DriftLossFraction == 0.0)) {
-                // assume Drift loss not entered and should be defaulted
-                tower.DriftLossFraction = 0.008 / 100.0;
-            }
```
Comment (Nigusse) [inline src/EnergyPlus/CondenserLoopTowers.cc:1767]: Deleted redundant code since the `Min-Fields` include this field.

**Review by Nigusse — COMMENTED** [review]
Extended the `Min-Fields` such that the input processor fills in the new default values added.

**Comment by Nigusse** [comment]
Results based on defect file UnitarySystem_MultiSpeedCoils_SingleMode_Mod2.idf

SizingResults_Test2

In the branch the `Design Size Rated Air Flow Rate [m3/s]` value is now reported for Speed 1, 2 and 3 because the flow rate is different by more than 10% of the user specified value. But the `Design Size Gross Cooling Capacity [W]` value is now not reported for Speed 1, 2 and 3 because the cooling capacity is within 10% of the user specified value.

**Comment by Nigusse** [comment]
Results based on defect file UnitarySystem_MultiSpeedCoils_SingleMode_Mod3.idf

SizingResults_Test3

For the third test case both the `Design Size Rated Air Flow Rate and Gross Cooling Capacity` are reported for each speed  because they differ by more than 10% from the user specified values. The scaling of the design size rated air flow rates and gross cooling capacities for each speed are based on the autosize value if available or user specified value if not.

File: idd/Energy+.idd.in [hunk 16]
```diff
@@ -75329,6 +75335,7 @@ CoolingTower:VariableSpeed,
        \type choice
        \key LossFactor
        \key SaturatedExit
+       \default SaturatedExit
```
Comment (mjwitte) [inline idd/Energy+.idd.in:75338]: CoolingTower:VariableSpeed still needs \default for Drift Loss Percent.

**Review by mjwitte — COMMENTED** [review]
@Nigusse Code and IDD changes look good - nice use of `GetEnumerationValue` and `format`.

Just one more IDD change needed.

Also, does the I/O Ref need any changes related to this?

File: idd/Energy+.idd.in (hunk 16 above)
Comment (Nigusse) [reply idd/Energy+.idd.in:75338]: OK. I will address this and check the I/O ref documentations.

**Comment by Nigusse** [comment]
The `Cell Control` input field default value for the four cooling tower models in the `IDD` is `MinimalCell` but in the code it is set to `MaximalCell` [here](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/CondenserLoopTowers.cc#L519-L540) for single speed cooling tower and [here](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/CondenserLoopTowers.cc#L1512-L1533) for variable speed cooling tower. *The default value in the IDD needs to change to MaximalCell to match what is in the code or vice versa.* what do you think? @mjwitte

**Comment by Nigusse** [comment]
@mjwitte Added default value of `0.008%` to `Drift Loss Percent` input field of `CoolingTower:VariableSpeed` object in the IDD and I/O Ref. Also did some minor format changes in the I/O Ref of cooling towers.

**Comment by Nigusse** [comment]
Numeric diffs (e.g. Water Use Volume) are expected in `ASHRAE901_Hospital_STD2019_Denver.idf` and `ASHRAE901_OfficeLarge_STD2019_Denver.idf` example files due to default value change for `Drift Loss Percent` input field from `0.0` to `0.008%`. For these two example idfs this field is blank. There are 12 example files with VS cooling tower, but the other 10 idfs have user specified value for this field.

File: doc/input-output-reference/src/overview/group-condenser-equipment.tex [hunk 17]
```diff
@@ -655,7 +655,7 @@ \subsubsection{Inputs}\label{inputs-1-006}
 
 \paragraph{Field: Evaporation Loss Mode}\label{field-evaporation-loss-mode-1}
 
-This field is used to choose which method is used to model the amount of water evaporated by the cooling tower. There are two options: `LossFactor' or `SaturatedExit'. The default is SaturatedExit. The user-defined loss factor is entered in the following field. By assuming that the air leaving the tower is saturated, the evaporation can be directly calculated using moist air engineering calculations with data available within the cooling tower model (and does not require additional user input).
+This field is used to choose which method is used to model the amount of water evaporated by the cooling tower. There are two options: \textbf{LossFactor} or \textbf{SaturatedExit}. The default is SaturatedExit. The user-defined loss factor is entered in the following field. By assuming that the air leaving the tower is saturated, the evaporation can be directly calculated using moist air engineering calculations with data available within the cooling tower model (and does not require additional user input).
```
Comment (Nigusse) [inline doc/input-output-reference/src/overview/group-condenser-equipment.tex:660]: Format change for consistency.

**Review by Nigusse — COMMENTED** [review]
Updated IDD and I/O Ref of VS cooling tower object.

**Comment by mjwitte** [comment]
> The `Cell Control` input field default value for the four cooling tower models in the `IDD` is `MinimalCell` but in the code it is set to `MaximalCell` [here](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/CondenserLoopTowers.cc#L519-L540) for single speed cooling tower and [here](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/CondenserLoopTowers.cc#L1512-L1533) for variable speed cooling tower. _The default value in the IDD needs to change to MaximalCell to match what is in the code or vice versa._ what do you think? @mjwitte

Well, that's messy. Looking back in the archive, when multi-cell towers were first added, all four towers had MinimalCell as the IDD default, and MaximalCell as the code default. And it's still that way, all four towers use MaximalCell if the field is blank. I would change the IDD defaults to match what the code has been doing all along. So, all four towers should have \default MaximalCell.

**Comment by Nigusse** [comment]
@mjwitte Thanks. I will make change to the IDD.

File: doc/input-output-reference/src/overview/group-condenser-equipment.tex [hunk 18]
```diff
@@ -193,6 +193,8 @@ \subsubsection{Inputs}\label{inputs-007}
 
 \textbf{MaximalCell}: As many cells as possible will be turned on. In no case, however, will the flow per cell be allowed to drop below its minimum value specified by the \emph{Minimum Water Flow Rate Fraction}.
 
+If this field is left blank, the default value is \textit{MaximalCell}.
+
```
Comment (Nigusse) [inline doc/input-output-reference/src/overview/group-condenser-equipment.tex:197]: Updated I/O Ref by adding default value for `Cell Control` input fields.

**Review by Nigusse — COMMENTED** [review]
Changed struct default value of `Cell Control` field for cooling tower in the Idd and I/O Ref and added Enum class for `Cell Control` choice input field.

**Comment by Nigusse** [comment]
@mjwitte I have addressed your comments and it is ready for your final review.

File: doc/input-output-reference/src/overview/group-condenser-equipment.tex [hunk 19]
```diff
@@ -697,6 +699,8 @@ \subsubsection{Inputs}\label{inputs-1-006}
...
```
Comment (Nigusse) [inline doc/input-output-reference/src/overview/group-condenser-equipment.tex:703]: Same as above.

File: doc/input-output-reference/src/overview/group-condenser-equipment.tex [hunk 20]
```diff
@@ -1209,6 +1213,8 @@ \subsubsection{Inputs}\label{inputs-2-006}
...
```
Comment (Nigusse) [inline doc/input-output-reference/src/overview/group-condenser-equipment.tex:1217]: Same as above.

File: src/EnergyPlus/CondenserLoopTowers.hh [hunk 21]
```diff
@@ -219,16 +219,16 @@ namespace CondenserLoopTowers {
         //  through the tower sump
         // multi cell tower
         int NumCell = 0; // Number of cells in the cooling tower
-        CellCtrl CellCtrl_Num = CellCtrl::Invalid;
+        CellCtrl CellCtrl_Num = CellCtrl::MaxCell;
```
Comment (Nigusse) [inline src/EnergyPlus/CondenserLoopTowers.hh]: Changed struct `CellCtrl_Num` member variable to the default value of `CellCtrl::MaxCell`.

File: src/EnergyPlus/CondenserLoopTowers.hh [hunk 22]
```diff
@@ -219,16 +219,16 @@ namespace CondenserLoopTowers {
...
         int NumCellOn = 0;            // number of cells working
         Real64 MinFracFlowRate = 0.0; // Minimal fraction of design flow/cell allowable
         Real64 MaxFracFlowRate = 0.0; // Maximal ratio of design flow/cell allowable
         // begin water system interactions
-        EvapLoss EvapLossMode = EvapLoss::MoistTheory; // sets how tower water evaporation is modeled
-        Real64 UserEvapLossFactor = 0.0;               // simple model [%/Delt C]
-        Real64 DriftLossFraction = 0.0;
+        EvapLoss EvapLossMode = EvapLoss::MoistTheory;   // sets how tower water evaporation is modeled
+        Real64 UserEvapLossFactor = 0.0;                 // simple model [%/Delt C]
+        Real64 DriftLossFraction = 0.008;                // default value is 0.008%
```
Comment (Nigusse) [inline src/EnergyPlus/CondenserLoopTowers.hh:229]: Changed initialization to IDD default value of `0.0008%` for `DriftLossFraction` from `0.0`.

File: src/EnergyPlus/CondenserLoopTowers.cc [hunk 23]
```diff
@@ -516,27 +495,9 @@ namespace CondenserLoopTowers {
...
-                    } else {
-                        ShowSevereError(state, "Illegal " + state.dataIPShortCut->cAlphaFieldNames(12) + " = " + AlphArray(12));
-                        ShowContinueError(state,
-                                          format("Occurs in {}={}", DataPlant::PlantEquipTypeNames[static_cast<int>(tower.TowerType)], tower.Name));
-                        ErrorsFound = true;
-                    }
-                }
-            } else {
-                // assume Cell Control not entered and should be defaulted
-                tower.CellCtrl_Num = CellCtrl::MaxCell;
```
Comment (Nigusse) [inline src/EnergyPlus/CondenserLoopTowers.cc:539]: Modified the code to use `getEnumerationValue` for `cell control` input field.

File: src/EnergyPlus/CondenserLoopTowers.cc [hunk 24]
```diff
@@ -847,27 +786,9 @@ namespace CondenserLoopTowers {
...
```
Comment (Nigusse) [inline src/EnergyPlus/CondenserLoopTowers.cc:870]: Same as above.

File: src/EnergyPlus/CondenserLoopTowers.cc [hunk 25]
```diff
@@ -1509,27 +1410,9 @@ namespace CondenserLoopTowers {
...
-                    } else {
-                        ShowSevereError(state, "Illegal " + state.dataIPShortCut->cAlphaFieldNames(13) + " = " + AlphArray(13));
-                        ShowContinueError(state,
-                                          format("Occurs in {}={}", DataPlant::PlantEquipTypeNames[static_cast<int>(tower.TowerType)], tower.Name));
-                        ErrorsFound = true;
-                    }
-                }
-            } else {
-                // assume Cell Control not entered and should be defaulted
-                tower.CellCtrl_Num = CellCtrl::MaxCell;
```
Comment (Nigusse) [inline src/EnergyPlus/CondenserLoopTowers.cc:1532]: Modified the code to use `getEnumerationValue` for `cell control` input field.

**Review by mjwitte — APPROVED** [review]
@Nigusse Looks good. I'll wait for CI to run, then merge.

**Comment by Nigusse** [comment]
I learned that in the process of addressing this issue. Replaced the member variable `CellCtrl_Num` with `cellCtrl` as suggested. @amirroth Thanks.

**Comment by Nigusse** [comment]
The `Cell Control` input field is not part of the `Min-Fields` count and it is for that reason that I added a check before calling `getEnumerationValue` for single speed cooling tower [here](https://github.com/NREL/EnergyPlus/blob/182646255_Issue9262/src/EnergyPlus/CondenserLoopTowers.cc#L498-L501) as a sample.

Input processor throws severe error message and fatal out if the input value is not valid. I did test each of the four cooling tower models. See below sample error message when invalid value is entered.
```
   ** Severe  ** <root>[CoolingTower:VariableSpeed:Merkel][TowerWaterSys CoolTower2][cell_control] - "AverageCell" - Failed to match against any enum values.
   **  Fatal  ** Errors occurred on processing input file. Preceding condition(s) cause termination.
```
So, adding `ErrorFound ||= (tower.cellCtrl == CellCtrl::Invalid;)` is not necessary, I think.

**Comment by Nigusse** [comment]
Sorry, it is the InputProcessor that throws the severe error message and fatal out for invalid enum Num, not the `getEnumerationValue`.

**Comment by joseph-robertson** [comment]
@mjwitte @Nigusse

What about making "Performance Input Method" for `EvaporativeFluidCooler:SingleSpeed` a required-field? See "Also" in https://github.com/NREL/EnergyPlus/issues/9262. (`EvaporativeFluidCooler:TwoSpeed` has required "Performance Input Method" already.)

**Comment by Nigusse** [comment]
@joseph-robertson you are correct, I missed to change the field as `required-field` for `EvaporativeFluidCooler:SingleSpeed`.  This can be addressed separately as a hot fix.

**Comment by mjwitte** [comment]
@Nigusse Do we want that field to be required or just assign a default? Does the code have a hidden default?

**Comment by Nigusse** [comment]
Yes, the field has to be required. If one of the three key choices is not entered, then the simulation will fatal out. No, the code does not have a hidden default.

**Pull request merged**